packages/react-badge/src/components/CounterBadge/renderCounterBadge.tsx

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import { getSlots } from '@fluentui/react-utilities';
+import { CounterBadgeState } from './CounterBadge.types';
+import { counterBadgeShorthandProps } from './useCounterBadge';
+
+export const renderCounterBadge = (state: CounterBadgeState) => {
+  const { slots, slotProps } = getSlots(state, counterBadgeShorthandProps);
+  let content = '';
+  if (state.count !== 0 || (state.count === 0 && state.showZero)) {
+    content = state.count > state.overflowCount ? `${state.overflowCount}+` : `${state.count}`;
+  }
```

Comment:
This type of logic should be done in `useCounterBadge` instead. You could set `state.children` to the text, and then call `renderBadge`, and skip creating `renderCounterBadge` entirely.

In fact, CounterBadge and PresenceBadge are great use cases for our new composition model.
* `useCounterBadge` should call `useBadge`, and then only make the updates to the state that are specific to `CounterBadge`
* `useCounterBadgeStyles` should call `useBadgeStyles`, then only add the extra styles specific to counter badge
*  only write a new render function if the structure of the slots is different (which isn't the case here)